react_app/src/reducers/game.reducers.js

```diff
@@ -0,0 +1,40 @@
+import * as ActionTypes from "actions/actionTypes";
+
+const initialState = {
+  teamName: "",
+  teamScore: 0,
+  teamRank: false,
+  viewingChallenge: null
+};
+
+/**
+ * action creators
+ */
+
+/**
+ * reducers
+ */
+/* state param takes default of initialState if not otherwise set */
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case ActionTypes.GET_GAME_STATE:
+      return {
+        ...state,                               // unpack state
```

Comment:
Let me know when you get sick of not being able to mutate `state` and I'll show you an awesome library so you can get around it.